### PR TITLE
[1770] Provider led only end-to-end feature spec

### PR DESCRIPTION
**Pull request by jhenderson**
### Context
https://trello.com/c/8SGXa7hI/1770-l-provider-led-only-end-to-end-feature-spec

### Changes proposed in this pull request
- New end-to-end feature spec that creates a provider-led trainee from scratch and fills in all the required information before submitting for TRN
- Tidy up up other feature specs - removing redundant steps, removing duplication etc

### CODE REVIEW

**Comment by jhenderson** [comment]
[[L] Provider-led only end to end feature spec](https://trello.com/c/8SGXa7hI/1770-l-provider-led-only-end-to-end-feature-spec)

File: spec/features/trainees/degrees/adding_degree_spec.rb [hunk 1]
```diff
@@ -156,11 +142,11 @@ def then_i_see_the_error_summary_for_degree_details_page
     expect(degree_details_page.error_summary).to be_visible
   end
 
-  def and_i_have_selected_uk_route
+  def given_i_have_selected_uk_route
     @locale = "uk"
   end
 
-  def and_i_have_selected_non_uk_route
+  def given_i_have_selected_non_uk_route
```
Comment (darokel) [inline spec/features/trainees/degrees/adding_degree_spec.rb:149]:
```suggestion
given_i_have_selected_the_non_uk_route
```

File: spec/features/trainees/edit_contact_details_spec.rb [hunk 2]
```diff
@@ -6,30 +6,30 @@
...
   end
 
+private
+
   def and_i_am_on_the_contact_details_page
     contact_details_page.load(id: trainee.slug)
   end
 
-  def and_i_enter_valid_parameters
+  def and_i_make_change
```
Comment (darokel) [inline spec/features/trainees/edit_contact_details_spec.rb:32]:
```suggestion
and_i_make_a_change
```

File: spec/support/page_objects/trainees/review_draft.rb [hunk 3]
```diff
@@ -17,6 +17,30 @@ class ReviewDraft < PageObjects::Base
 
       element :review_this_record_link, "#check-details"
       element :delete_this_draft_link, ".app-link--warning"
+
+      def has_personal_details_completed?
+        personal_details.status.text == "Status completed"
```
Comment (darokel) [inline spec/support/page_objects/trainees/review_draft.rb:22]: "Status completed" could probably go into a constant in this class

**Review by darokel — COMMENTED** [review]
Awesome, this is looking much better so far. Just a couple of tiny comments

**Review by darokel — APPROVED** [review]

**Pull request merged**